In [46]:
import pandas as pd

In [47]:
data=pd.read_csv('loan_prediction.csv')

# Loan_ID : Unique Loan ID

# Gender : Male/ Female

# Married : Applicant married (Y/N)

# Dependents : Number of dependents

# Education : Applicant Education (Graduate/ Under Graduate)

# Self_Employed : Self employed (Y/N)

# ApplicantIncome : Applicant income

# CoapplicantIncome : Coapplicant income

# LoanAmount : Loan amount in thousands of dollars

# Loan_Amount_Term : Term of loan in months

# Credit_History : Credit history meets guidelines yes or no

# Property_Area : Urban/ Semi Urban/ Rural

# Loan_Status : Loan approved (Y/N) this is the target variable


In [48]:
data.head()

Loan_ID Gender Married Dependents     Education Self_Employed  \
0  LP001002   Male      No          0      Graduate            No   
1  LP001003   Male     Yes          1      Graduate            No   
2  LP001005   Male     Yes          0      Graduate           Yes   
3  LP001006   Male     Yes          0  Not Graduate            No   
4  LP001008   Male      No          0      Graduate            No   

   ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0             5849                0.0         NaN             360.0   
1             4583             1508.0       128.0             360.0   
2             3000                0.0        66.0             360.0   
3             2583             2358.0       120.0             360.0   
4             6000                0.0       141.0             360.0   

   Credit_History Property_Area Loan_Status  
0             1.0         Urban           Y  
1             1.0         Rural           N  
2             1.0         Urban           Y  
3             1.0         Urban           Y  
4             1.0         Urban           Y

In [49]:
data.tail()

Loan_ID  Gender Married Dependents Education Self_Employed  \
609  LP002978  Female      No          0  Graduate            No   
610  LP002979    Male     Yes         3+  Graduate            No   
611  LP002983    Male     Yes          1  Graduate            No   
612  LP002984    Male     Yes          2  Graduate            No   
613  LP002990  Female      No          0  Graduate           Yes   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
609             2900                0.0        71.0             360.0   
610             4106                0.0        40.0             180.0   
611             8072              240.0       253.0             360.0   
612             7583                0.0       187.0             360.0   
613             4583                0.0       133.0             360.0   

     Credit_History Property_Area Loan_Status  
609             1.0         Rural           Y  
610             1.0         Rural           Y  
611             1.0         Urban           Y  
612             1.0         Urban           Y  
613             0.0     Semiurban           N

In [50]:
data.shape
print("Number of Rows:",data.shape[0])
print("Number of Columns:",data.shape[1])

Number of Rows: 614
Number of Columns: 13


Datatype of column

In [51]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    object 
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    object 
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(4), int64(1), object(8)
memory usage: 62.5+ KB


Check Null Values In The Dataset

In [52]:
data.isnull()

Loan_ID  Gender  Married  Dependents  Education  Self_Employed  \
0      False   False    False       False      False          False   
1      False   False    False       False      False          False   
2      False   False    False       False      False          False   
3      False   False    False       False      False          False   
4      False   False    False       False      False          False   
..       ...     ...      ...         ...        ...            ...   
609    False   False    False       False      False          False   
610    False   False    False       False      False          False   
611    False   False    False       False      False          False   
612    False   False    False       False      False          False   
613    False   False    False       False      False          False   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0              False              False        True             False   
1              False              False       False             False   
2              False              False       False             False   
3              False              False       False             False   
4              False              False       False             False   
..               ...                ...         ...               ...   
609            False              False       False             False   
610            False              False       False             False   
611            False              False       False             False   
612            False              False       False             False   
613            False              False       False             False   

     Credit_History  Property_Area  Loan_Status  
0             False          False        False  
1             False          False        False  
2             False          False        False  
3             False          False        False  
4             False          False        False  
..              ...            ...          ...  
609           False          False        False  
610           False          False        False  
611           False          False        False  
612           False          False        False  
613           False          False        False  

[614 rows x 13 columns]

In [53]:
data.isnull().sum()

Loan_ID               0
Gender               13
Married               3
Dependents           15
Education             0
Self_Employed        32
ApplicantIncome       0
CoapplicantIncome     0
LoanAmount           22
Loan_Amount_Term     14
Credit_History       50
Property_Area         0
Loan_Status           0
dtype: int64

In [54]:
data.isnull().sum()*100/len(data)

Loan_ID              0.000000
Gender               2.117264
Married              0.488599
Dependents           2.442997
Education            0.000000
Self_Employed        5.211726
ApplicantIncome      0.000000
CoapplicantIncome    0.000000
LoanAmount           3.583062
Loan_Amount_Term     2.280130
Credit_History       8.143322
Property_Area        0.000000
Loan_Status          0.000000
dtype: float64

Handling the missing value ( By drop the values that are less than 5% )

In [55]:
data = data.drop('Loan_ID',axis=1)

In [56]:
columns = ['Gender','Dependents','LoanAmount','Loan_Amount_Term']

In [57]:
data=data.dropna(subset=columns)  # dropna() = used to remove missing value

In [58]:
data.isnull().sum()*100/len(data)

Gender               0.000000
Married              0.000000
Dependents           0.000000
Education            0.000000
Self_Employed        5.424955
ApplicantIncome      0.000000
CoapplicantIncome    0.000000
LoanAmount           0.000000
Loan_Amount_Term     0.000000
Credit_History       8.679928
Property_Area        0.000000
Loan_Status          0.000000
dtype: float64

Fill the missing value for greater than 5%

In [59]:
data['Self_Employed'].mode()[0]   # mode() = use to display the most frequently used data

'No'

In [60]:
data['Self_Employed'].fillna(data['Self_Employed'].mode()[0])

1       No
2      Yes
3       No
4       No
5      Yes
      ... 
609     No
610     No
611     No
612     No
613    Yes
Name: Self_Employed, Length: 553, dtype: object

In [61]:
data['Self_Employed']=data['Self_Employed'].fillna(data['Self_Employed'].mode()[0])

In [62]:
data.isnull().sum()*100/len(data)

Gender               0.000000
Married              0.000000
Dependents           0.000000
Education            0.000000
Self_Employed        0.000000
ApplicantIncome      0.000000
CoapplicantIncome    0.000000
LoanAmount           0.000000
Loan_Amount_Term     0.000000
Credit_History       8.679928
Property_Area        0.000000
Loan_Status          0.000000
dtype: float64

In [63]:
data['Credit_History'].unique()

array([ 1.,  0., nan])

In [64]:
data['Credit_History'].fillna(data['Credit_History'].mode()[0])

1      1.0
2      1.0
3      1.0
4      1.0
5      1.0
      ... 
609    1.0
610    1.0
611    1.0
612    1.0
613    0.0
Name: Credit_History, Length: 553, dtype: float64

In [65]:
data['Credit_History']=data['Credit_History'].fillna(data['Credit_History'].mode()[0])

In [66]:
data.isnull().sum()*100/len(data)

Gender               0.0
Married              0.0
Dependents           0.0
Education            0.0
Self_Employed        0.0
ApplicantIncome      0.0
CoapplicantIncome    0.0
LoanAmount           0.0
Loan_Amount_Term     0.0
Credit_History       0.0
Property_Area        0.0
Loan_Status          0.0
dtype: float64

Handling Categorical Columns

In [67]:
data.sample(10)

Gender Married Dependents     Education Self_Employed  ApplicantIncome  \
461    Male     Yes         3+      Graduate            No             7740   
174    Male     Yes          0  Not Graduate           Yes             4344   
156    Male     Yes          1      Graduate            No             6000   
290    Male     Yes          0      Graduate            No             3075   
211    Male     Yes         3+      Graduate            No             3430   
146  Female     Yes          2      Graduate            No            14866   
279  Female     Yes          0  Not Graduate            No             4100   
386    Male      No          0  Not Graduate            No             2346   
10     Male     Yes          2      Graduate            No             3200   
281    Male     Yes          0      Graduate            No             3927   

     CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \
461                0.0       128.0             180.0             1.0   
174              736.0        87.0             360.0             1.0   
156                0.0       160.0             360.0             1.0   
290             2416.0       139.0             360.0             1.0   
211             1250.0       128.0             360.0             0.0   
146                0.0        70.0             360.0             1.0   
279                0.0       124.0             360.0             1.0   
386             1600.0       132.0             360.0             1.0   
10               700.0        70.0             360.0             1.0   
281              800.0       112.0             360.0             1.0   

    Property_Area Loan_Status  
461         Urban           Y  
174     Semiurban           N  
156         Rural           Y  
290         Rural           Y  
211     Semiurban           N  
146         Urban           Y  
279         Rural           Y  
386     Semiurban           Y  
10          Urban           Y  
281     Semiurban           Y

in dependents column i have value:"3+", tO REPLACE to "4"

In [68]:
data['Dependents'].replace(to_replace="3+",value='4')

1      1
2      0
3      0
4      0
5      2
      ..
609    0
610    4
611    1
612    2
613    0
Name: Dependents, Length: 553, dtype: object

In [69]:
data['Dependents']= data['Dependents'].replace(to_replace="3+",value='4')

In [70]:
data['Dependents'].unique()

array(['1', '0', '2', '4'], dtype=object)

As mechine learning algo. understand only numberical value..so convert it by using "MAP" function

In [71]:
data['Gender'].unique()

array(['Male', 'Female'], dtype=object)

In [72]:
data['Gender']=data['Gender'].map({'Male':1,'Female':0}).astype('int')

In [73]:
data['Married']=data['Married'].map({'Yes':1,'No':0}).astype('int')

In [74]:
data['Education']=data['Education'].map({'Graduate':1,'Not Graduate':0}).astype('int')

In [75]:
data['Self_Employed']=data['Self_Employed'].map({'Yes':1,'No':0}).astype('int')

In [76]:
data['Property_Area']=data['Property_Area'].map({'Rural':0,'Urban':1,'Semiurban':2}).astype('int')

In [77]:
data['Loan_Status']=data['Loan_Status'].map({'Y':1,'N':0}).astype('int')

In [78]:
data.head()

Gender  Married Dependents  Education  Self_Employed  ApplicantIncome  \
1       1        1          1          1              0             4583   
2       1        1          0          1              1             3000   
3       1        1          0          0              0             2583   
4       1        0          0          1              0             6000   
5       1        1          2          1              1             5417   

   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \
1             1508.0       128.0             360.0             1.0   
2                0.0        66.0             360.0             1.0   
3             2358.0       120.0             360.0             1.0   
4                0.0       141.0             360.0             1.0   
5             4196.0       267.0             360.0             1.0   

   Property_Area  Loan_Status  
1              0            0  
2              1            1  
3              1            1  
4              1            1  
5              1            1

Store Feature Matrix In X And Response (Target) In Vector y

In [79]:
X=data.drop('Loan_Status',axis=1)

In [80]:
y=data['Loan_Status']


In [81]:
y

1      0
2      1
3      1
4      1
5      1
      ..
609    1
610    1
611    1
612    1
613    0
Name: Loan_Status, Length: 553, dtype: int32

**Feature Scaling** ( *it's essential for ML algo when calculate the distance between the data.. if not scale, its starts dominate the distances*)

In [82]:
cols=['ApplicantIncome','CoapplicantIncome','LoanAmount','Loan_Amount_Term']

In [83]:
from sklearn.preprocessing import StandardScaler
st = StandardScaler()
X[cols]=st.fit_transform(X[cols])

In [84]:
X

Gender  Married Dependents  Education  Self_Employed  ApplicantIncome  \
1         1        1          1          1              0        -0.128694   
2         1        1          0          1              1        -0.394296   
3         1        1          0          0              0        -0.464262   
4         1        0          0          1              0         0.109057   
5         1        1          2          1              1         0.011239   
..      ...      ...        ...        ...            ...              ...   
609       0        0          0          1              0        -0.411075   
610       1        1          4          1              0        -0.208727   
611       1        1          1          1              0         0.456706   
612       1        1          2          1              0         0.374659   
613       0        0          0          1              1        -0.128694   

     CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \
1            -0.049699   -0.214368          0.279961             1.0   
2            -0.545638   -0.952675          0.279961             1.0   
3             0.229842   -0.309634          0.279961             1.0   
4            -0.545638   -0.059562          0.279961             1.0   
5             0.834309    1.440866          0.279961             1.0   
..                 ...         ...               ...             ...   
609          -0.545638   -0.893134          0.279961             1.0   
610          -0.545638   -1.262287         -2.468292             1.0   
611          -0.466709    1.274152          0.279961             1.0   
612          -0.545638    0.488213          0.279961             1.0   
613          -0.545638   -0.154828          0.279961             0.0   

     Property_Area  
1                0  
2                1  
3                1  
4                1  
5                1  
..             ...  
609              0  
610              0  
611              1  
612              1  
613              2  

[553 rows x 11 columns]

Splitting The Dataset Into The Training Set And Test Set & Applying K-Fold Cross Validation

In [85]:
from sklearn.model_selection import train_test_split
from sklearn.model_selection import cross_val_score
from sklearn.metrics import accuracy_score
import numpy as np

*create function to perform the task*

In [86]:
model_df={}
def model_val(model,X,y):
    X_train,X_test,y_train,y_test=train_test_split(X,y,
                                                   test_size=0.20,
                                                   random_state=42)
    model.fit(X_train,y_train)
    y_pred=model.predict(X_test)
    print(f"{model} accuracy is {accuracy_score(y_test,y_pred)}")

    score = cross_val_score(model,X,y,cv=5)
    print(f"{model} Avg cross val score is {np.mean(score)}")
    model_df[model]=round(np.mean(score)*100,2)

In [87]:
model_df

{}

**Logistic Regression**

In [88]:
from sklearn.linear_model import LogisticRegression
model = LogisticRegression()
model_val(model,X,y)

LogisticRegression() accuracy is 0.8018018018018018
LogisticRegression() Avg cross val score is 0.8047829647829647


**SVC**

In [89]:
from sklearn import svm
model = svm.SVC()
model_val(model,X,y)

SVC() accuracy is 0.7927927927927928
SVC() Avg cross val score is 0.7938902538902539


**Decision Tree Classifier**

In [90]:
from sklearn.tree import DecisionTreeClassifier
model = DecisionTreeClassifier()
model_val(model,X,y)

DecisionTreeClassifier() accuracy is 0.7657657657657657
DecisionTreeClassifier() Avg cross val score is 0.7143652743652743


**Random Forest Classifier**

In [91]:
from sklearn.ensemble import RandomForestClassifier
model=RandomForestClassifier()
model_val(model,X,y)

RandomForestClassifier() accuracy is 0.7747747747747747
RandomForestClassifier() Avg cross val score is 0.7776248976248976


In [92]:
model_df

{LogisticRegression(): 80.48,
 SVC(): 79.39,
 DecisionTreeClassifier(): 71.44,
 RandomForestClassifier(): 77.76}

**Hyperparamater Tuning**

In [93]:
from sklearn.model_selection import RandomizedSearchCV

**HP.T =** *Logistic Regression*

In [94]:
log_reg_grid={"C":np.logspace(-4,4,20),
              "solver":['liblinear']}

In [95]:
rs_log_reg=RandomizedSearchCV(LogisticRegression(),
                   param_distributions=log_reg_grid,
                   n_iter=20,cv=5,verbose=True)

In [96]:
rs_log_reg.fit(X,y)

Fitting 5 folds for each of 20 candidates, totalling 100 fits


RandomizedSearchCV(cv=5, estimator=LogisticRegression(), n_iter=20,
                   param_distributions={'C': array([1.00000000e-04, 2.63665090e-04, 6.95192796e-04, 1.83298071e-03,
       4.83293024e-03, 1.27427499e-02, 3.35981829e-02, 8.85866790e-02,
       2.33572147e-01, 6.15848211e-01, 1.62377674e+00, 4.28133240e+00,
       1.12883789e+01, 2.97635144e+01, 7.84759970e+01, 2.06913808e+02,
       5.45559478e+02, 1.43844989e+03, 3.79269019e+03, 1.00000000e+04]),
                                        'solver': ['liblinear']},
                   verbose=True)

In [97]:
rs_log_reg.best_score_

0.8047829647829647

In [98]:
rs_log_reg.best_params_

{'solver': 'liblinear', 'C': 0.23357214690901212}

**HP.T =** *SVC*

In [99]:
svc_grid={'C':[0.25,0.50,0.75,1],"kernel":["linear"]}

In [100]:
rs_svc=RandomizedSearchCV(svm.SVC(),
                   param_distributions=svc_grid,
                   cv=5,
                   n_iter=20,verbose=True)

In [101]:
rs_svc.fit(X,y)

c:\Users\jayas\AppData\Local\Programs\Python\Python311\Lib\site-packages\sklearn\model_selection\_search.py:320: UserWarning: The total space of parameters 4 is smaller than n_iter=20. Running 4 iterations. For exhaustive searches, use GridSearchCV.
  warnings.warn(


Fitting 5 folds for each of 4 candidates, totalling 20 fits


RandomizedSearchCV(cv=5, estimator=SVC(), n_iter=20,
                   param_distributions={'C': [0.25, 0.5, 0.75, 1],
                                        'kernel': ['linear']},
                   verbose=True)

In [102]:
rs_svc.best_score_

0.8066011466011467

In [103]:
rs_svc.best_params_

{'kernel': 'linear', 'C': 0.25}

**HP.T =** *Random Forest Classifier*

In [104]:
RandomForestClassifier()

RandomForestClassifier()

In [105]:
rf_grid={'n_estimators':np.arange(10,1000,10),
  'max_features':['auto','sqrt'],
 'max_depth':[None,3,5,10,20,30],
 'min_samples_split':[2,5,20,50,100],
 'min_samples_leaf':[1,2,5,10]
 }

In [106]:
rs_rf=RandomizedSearchCV(RandomForestClassifier(),
                  param_distributions=rf_grid,
                   cv=5,
                   n_iter=20,
                  verbose=True)

In [107]:
rs_rf.fit(X,y)

Fitting 5 folds for each of 20 candidates, totalling 100 fits


c:\Users\jayas\AppData\Local\Programs\Python\Python311\Lib\site-packages\sklearn\model_selection\_validation.py:540: FitFailedWarning: 
50 fits failed out of a total of 100.
The score on these train-test partitions for these parameters will be set to nan.
If these failures are not expected, you can try to debug them by setting error_score='raise'.

Below are more details about the failures:
--------------------------------------------------------------------------------
50 fits failed with the following error:
Traceback (most recent call last):
  File "c:\Users\jayas\AppData\Local\Programs\Python\Python311\Lib\site-packages\sklearn\model_selection\_validation.py", line 888, in _fit_and_score
    estimator.fit(X_train, y_train, **fit_params)
  File "c:\Users\jayas\AppData\Local\Programs\Python\Python311\Lib\site-packages\sklearn\base.py", line 1466, in wrapper
    estimator._validate_params()
  File "c:\Users\jayas\AppData\Local\Programs\Python\Python311\Lib\site-packages\sklearn\base.p

RandomizedSearchCV(cv=5, estimator=RandomForestClassifier(), n_iter=20,
                   param_distributions={'max_depth': [None, 3, 5, 10, 20, 30],
                                        'max_features': ['auto', 'sqrt'],
                                        'min_samples_leaf': [1, 2, 5, 10],
                                        'min_samples_split': [2, 5, 20, 50,
                                                              100],
                                        'n_estimators': array([ 10,  20,  30,  40,  50,  60,  70,  80,  90, 100, 110, 120, 130,
       140, 150, 160, 170, 180, 190, 200, 210, 220, 230, 240, 250, 260,
       270, 280, 290, 300, 310, 320, 330, 340, 350, 360, 370, 380, 390,
       400, 410, 420, 430, 440, 450, 460, 470, 480, 490, 500, 510, 520,
       530, 540, 550, 560, 570, 580, 590, 600, 610, 620, 630, 640, 650,
       660, 670, 680, 690, 700, 710, 720, 730, 740, 750, 760, 770, 780,
       790, 800, 810, 820, 830, 840, 850, 860, 870, 880, 890, 900, 910,
       920, 930, 940, 950, 960, 970, 980, 990])},
                   verbose=True)

In [108]:
rs_rf.best_score_

0.8066011466011467

In [109]:
rs_rf.best_params_

{'n_estimators': 490,
 'min_samples_split': 20,
 'min_samples_leaf': 5,
 'max_features': 'sqrt',
 'max_depth': 5}

*LogisticRegression score Before Hyperparameter Tuning: 80.48*

*LogisticRegression score after Hyperparameter Tuning: 80.48*

*SVC score Before Hyperparameter Tuning: 79.38*

*SVC score after Hyperparameter Tuning: 80.66*

*RandomForestClassifier score Before Hyperparameter Tuning: 77.76*

*RandomForestClassifier score after Hyperparameter Tuning: 80.66*

**Save The Model**

In [110]:
X=data.drop('Loan_Status',axis=1)
y=data['Loan_Status']

In [113]:
rf= RandomForestClassifier(n_estimators = 150,
    min_samples_split = 20,
    min_samples_leaf = 10,
    max_features = 'sqrt',
    max_depth = 3)

In [114]:
rf.fit(X,y)

RandomForestClassifier(max_depth=3, min_samples_leaf=10, min_samples_split=20,
                       n_estimators=150)

In [115]:
import joblib

In [116]:
joblib.dump(rf,'loan_status_predict')

['loan_status_predict']

In [117]:
model=joblib.load('loan_status_predict')

In [118]:
import pandas as pd
df = pd.DataFrame({
    'Gender':1,
    'Married':1,
    'Dependents':2,
    'Education':0,
    'Self_Employed':0,
    'ApplicantIncome':2889,
    'CoapplicantIncome':0.0,
    'LoanAmount':45,
    'Loan_Amount_Term':180,
    'Credit_History':0,
    'Property_Area':1
},index=[0])

In [119]:
df

Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \
0       1        1           2          0              0             2889   

   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \
0                0.0          45               180               0   

   Property_Area  
0              1

In [120]:
result = model.predict(df)

In [121]:
if result == 1:
  print("Loan Approved")
else:
  print("Loan Not Approved")

Loan Not Approved


**GUI**

In [122]:
from tkinter import *
import joblib
import pandas as pd

In [1]:
import joblib
import pandas as pd
from tkinter import *

def show_entry():
    p1 = float(e1.get())
    p2 = float(e2.get())
    p3 = float(e3.get())
    p4 = float(e4.get())
    p5 = float(e5.get())
    p6 = float(e6.get())
    p7 = float(e7.get())
    p8 = float(e8.get())
    p9 = float(e9.get())
    p10 = float(e10.get())
    p11 = float(e11.get())

    model = joblib.load('loan_status_predict')
    df = pd.DataFrame({
        'Gender': [p1],
        'Married': [p2],
        'Dependents': [p3],
        'Education': [p4],
        'Self_Employed': [p5],
        'ApplicantIncome': [p6],
        'CoapplicantIncome': [p7],
        'LoanAmount': [p8],
        'Loan_Amount_Term': [p9],
        'Credit_History': [p10],
        'Property_Area': [p11]
    })
    result = model.predict(df)

    if result == 1:
        Label(master, text="Loan Approved").grid(row=31)
    else:
        Label(master, text="Loan Not Approved").grid(row=31)

master = Tk()
master.geometry("400x900")
master.title("Loan Status Prediction Using Machine Learning")

Label(master, text="Loan Status Prediction", bg="black", fg="white").grid(row=0, columnspan=2)
Label(master, text="Gender [1:Male, 0:Female]").grid(row=1)
Label(master, text="Married [1:Yes, 0:No]").grid(row=2)
Label(master, text="Dependents [1,2,3,4]").grid(row=3)
Label(master, text="Education").grid(row=4)
Label(master, text="Self_Employed").grid(row=5)
Label(master, text="ApplicantIncome").grid(row=6)
Label(master, text="CoapplicantIncome").grid(row=7)
Label(master, text="LoanAmount").grid(row=8)
Label(master, text="Loan_Amount_Term").grid(row=9)
Label(master, text="Credit_History").grid(row=10)
Label(master, text="Property_Area").grid(row=11)

e1 = Entry(master)
e2 = Entry(master)
e3 = Entry(master)
e4 = Entry(master)
e5 = Entry(master)
e6 = Entry(master)
e7 = Entry(master)
e8 = Entry(master)
e9 = Entry(master)
e10 = Entry(master)
e11 = Entry(master)

e1.grid(row=1, column=1)
e2.grid(row=2, column=1)
e3.grid(row=3, column=1)
e4.grid(row=4, column=1)
e5.grid(row=5, column=1)
e6.grid(row=6, column=1)
e7.grid(row=7, column=1)
e8.grid(row=8, column=1)
e9.grid(row=9, column=1)
e10.grid(row=10, column=1)
e11.grid(row=11, column=1)

Button(master, text="Predict", command=show_entry).grid()

master.mainloop()
